common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -1341,11 +1345,17 @@ def create_child(self, user_id, parent_usage_key, block_type, block_id=None, **k
             block_id: a unique identifier for the new item. If not supplied,
                 a new identifier will be generated
         """
-        xblock = self.create_item(user_id, parent_usage_key.course_key, block_type, block_id=block_id, **kwargs)
         # attach to parent if given
-        if 'detached' not in xblock._class_tags:
+        if parent_usage_key is not None:
             parent = self.get_item(parent_usage_key)
+            kwargs.setdefault('for_parent', parent)
+
+        else:
+            parent = None
```

Comment:
I prefer adding a `parent = None` before the `if` clause and eliminating this `else` clause. Is there a coding standard involved here?